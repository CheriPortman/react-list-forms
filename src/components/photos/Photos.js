import React from 'react';
import PropTypes from 'prop-types';
import Photo from './Photo';

// take array of photos and create a photo for each
// display all Photo componenets in a ul

function Photos({ photos }) {
  const photoList = photos.map(photo => {
    return (
      <li key={photo}>  
        <Photo photo={photo} />
      </li>
    );
  });

  return (
    <ul>
      {photoList}
    </ul>
  );
}

Photos.propTypes = {
  photos: PropTypes.array.isRequired
};

export default Photos;
