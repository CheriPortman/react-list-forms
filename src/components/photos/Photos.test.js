import React from 'react';
import { shallow } from 'enzyme';
import Photos from './Photos';

describe('Photos Component', () => {
  it('renders photos', () => {
    const photos = [
      'src/assets/fanno-creek-fun.jpg',
      'src/assets/dragon-hoard.jpg',
      'src/assets/crater-lake-dan.jpg'
    ];
    const wrapper = shallow(<Photos photos={photos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
