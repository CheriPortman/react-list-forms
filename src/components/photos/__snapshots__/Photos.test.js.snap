// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos Component renders photos 1`] = `
<ul>
  <li
    key="src/assets/fanno-creek-fun.jpg"
  >
    <Photo
      photo="src/assets/fanno-creek-fun.jpg"
    />
  </li>
  <li
    key="src/assets/dragon-hoard.jpg"
  >
    <Photo
      photo="src/assets/dragon-hoard.jpg"
    />
  </li>
  <li
    key="src/assets/crater-lake-dan.jpg"
  >
    <Photo
      photo="src/assets/crater-lake-dan.jpg"
    />
  </li>
</ul>
`;
